galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.ceda.manager/src/main/java/dev/galasa/cicsts/ceda/internal/CedaImpl.java

```diff
@@ -7,32 +7,44 @@
 
 import javax.validation.constraints.NotNull;
 
-import dev.galasa.cicsts.ceda.CEDAException;
-import dev.galasa.cicsts.ceda.ICEDA;
+import dev.galasa.cicsts.CedaException;
+import dev.galasa.cicsts.ICeda;
+import dev.galasa.cicsts.ICicsRegion;
+import dev.galasa.cicsts.ICicsTerminal;
 import dev.galasa.zos3270.FieldNotFoundException;
-import dev.galasa.zos3270.ITerminal;
 import dev.galasa.zos3270.KeyboardLockedException;
 import dev.galasa.zos3270.TerminalInterruptedException;
 import dev.galasa.zos3270.TimeoutException;
 import dev.galasa.zos3270.spi.NetworkException;
 
-public class CEDAImpl implements ICEDA{
+public class CedaImpl implements ICeda{
 
-	private ITerminal terminal;
+	private ICicsRegion cicsRegion;
+	
+	private ICicsTerminal terminal;
+	
+	public CedaImpl(ICicsRegion cicsRegion) {
+		this.cicsRegion = cicsRegion;
+	}
+	
 
 	@Override
-	public void createResource(@NotNull ITerminal cedaTerminal, @NotNull String resourceType,
-			@NotNull String resourceName, @NotNull String groupName, String resourceParameters) throws CEDAException{
+	public void createResource(@NotNull ICicsTerminal terminal, @NotNull String resourceType, 
+            @NotNull String resourceName, @NotNull String groupName, String resourceParameters) throws CedaException{
+		
+		if(cicsRegion != terminal.getCicsRegion()) {
+			 throw new CedaException("Version collition");
```

Comment:
Message needs to be a bit more explicit,

The provided terminal is not from the correct CICS Region